crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -1165,8 +1164,18 @@ where
                             .is_none()
                         {
                             // Update the state and hashes of the blockchain tree if possible.
-                            match self.restore_tree_if_possible(sync_target_state) {
-                                Ok(_) => self.sync_state_updater.update_sync_state(SyncState::Idle),
+                            match self.update_tree_on_finished_pipeline(
+                                sync_target_state.finalized_block_hash,
+                            ) {
+                                Ok(synced) => {
+                                    if !synced {
+                                        // // We don't have the finalized block in the database, so
+                                        // we need to run another pipeline.
+                                        self.sync.set_pipeline_sync_target(
+                                            sync_target_state.finalized_block_hash,
```

Comment:
what's the motivation behind this change? we cannot add blocks on top of finalized unless they are in the block buffer. that potentially means more pipeline runs before live sync

Reply:
Running the pipeline with the finalized hash seems safer than with the head at least, I don't really see an issue with this